Sources/GDSCommon/README.md

```diff
@@ -122,6 +122,84 @@ loadResourceWithCallback() {
 
 ## Patterns
 
+### BaseViewController
+
+``BaseViewController`` inherits from `UIViewController`. ``BaseViewController`` includes the repetitive code that all screens should have to avoid repetition and reduce risk of missing important functionality and to make it easier to amend, or fix defects if they arise.
+
+Screen view controllers should generally inherit from ``BaseViewController`` instead of `UIViewController` unless the functionality of the screen needs to be intentionally different from standard screens.
+
+#### To use:
+
+Once inheriting from ``BaseViewController`` and adding conformance to ``BaseViewModel`` to concrete view model, in the future do not add the following properties and methods to view model protocols:
```

Comment:
It feels like this documentation is fulfilling two purposes:
1. Explaining how the package is built to our internal developers
2. Explaining how to use the package to the consumers of the package

I think the `README` should focus on the latter, and we should have additional documentation for our internal folks, as needed.